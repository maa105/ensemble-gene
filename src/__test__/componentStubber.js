import _ from 'lodash';

var lifecycleMethods = [
  'render',
  'componentWillMount',
  'componentDidMount',
  'componentWillReceiveProps',
  'shouldComponentUpdate',
  'componentWillUpdate',
  'componentDidUpdate',
  'componentWillUnmount'
];

var stubComponent = function(componentClass) {
  var originalPropTypes, mocked = [];

  beforeEach(function() {
    originalPropTypes = componentClass.propTypes;

    componentClass.propTypes = {};

    _.forEach(lifecycleMethods, function(method) {
      if(typeof componentClass.prototype[method] !== 'undefined') {
        mocked.push(method);
        const spy = jest.spyOn(componentClass.prototype, method).mockImplementation(() => null);
        componentClass.prototype[method].spy = spy;
      }
    });
  });

  afterEach(function() {
    componentClass.propTypes = originalPropTypes;
    _.forEach(mocked, function(method) {
      componentClass.prototype[method].spy.mockRestore();
    });
  });
};

module.exports = {
  stubComponent: stubComponent
};
