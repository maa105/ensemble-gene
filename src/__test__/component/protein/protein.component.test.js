import React from 'react';
import ProteinSequence from '../../../components/protein/protein-sequence.component';
import Protein from '../../../components/protein/protein.component';
import renderer from 'react-test-renderer';
import { stubComponent } from '../../componentStubber';

describe('protien component test', function() {
  stubComponent(ProteinSequence);
  
  const props = { 
    protein: {
      proteinId: 'proteinId',
      seq: 'seq'
    }, 
    defaultExpanded: true,
    position: 600,
    check: 'A',
    replacement: 'V',
    eventKey: 2
  };

  test('Protein with proper props to render correctly', () => {
    const component = renderer.create(
      <Protein {...props} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const sequences = component.root.findAllByType(ProteinSequence);
    expect(sequences.length).toBe(1);
    
    const sequenceProps = sequences[0].props;

    // props passed down properly to Protein sequence
    expect(sequenceProps).toEqual({ sequence: 'seq', position: 600, check: 'A', replacement: 'V' });

  });
});
