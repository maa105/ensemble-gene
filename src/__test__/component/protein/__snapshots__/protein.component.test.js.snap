// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`protien component test Protein with proper props to render correctly 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
    id="2--heading"
    role="tab"
  >
    <div
      className="panel-title"
    >
      <a
        aria-controls="2--body"
        aria-expanded={true}
        className=""
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      />
    </div>
  </div>
  <div
    aria-expanded={true}
    aria-labelledby="2--heading"
    className="panel-collapse collapse in"
    id="2--body"
    role="tabpanel"
  >
    <div
      className="panel-body"
    />
  </div>
</div>
`;
